ddl/ddl_api.go

```diff
@@ -1029,6 +1026,51 @@ func (d *ddl) CreateTableWithLike(ctx sessionctx.Context, ident, referIdent ast.
 	return errors.Trace(err)
 }
 
+func buildTableInfoWithLike(ident ast.Ident, referTblInfo *model.TableInfo) model.TableInfo {
+	tblInfo := *referTblInfo
+	// Check non-public column and adjust column offset.
+	newColumns := make([]*model.ColumnInfo, 0, len(tblInfo.Columns))
+	offsetChanged := make(map[int]int)
+	for _, col := range tblInfo.Columns {
+		if col.State == model.StatePublic {
+			newCol := col.Clone()
```

Comment:
If `len(tbl.Cols()) == len(tblInfo.Columns)` is true, I think we need not clone columns.

Reply:
done.